File: pdu/python/hako_msgs/pdu_cdr_conv_DisturbanceWind.py
from .pdu_pytype_DisturbanceWind import DisturbanceWind
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_DisturbanceWind(writer: CdrWriter, src: DisturbanceWind):
    py_to_cdr_body_Vector3(writer, src.value)


def cdr_body_to_py_DisturbanceWind(reader: CdrReader, dst: DisturbanceWind):
    cdr_body_to_py_Vector3(reader, dst.value)
    return dst


def py_to_cdr_DisturbanceWind(src: DisturbanceWind) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DisturbanceWind(writer, src)
    return writer.bytes()


def cdr_to_py_DisturbanceWind(cdr_payload) -> DisturbanceWind:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DisturbanceWind()
    return cdr_body_to_py_DisturbanceWind(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_DisturbanceWind.js
import { DisturbanceWind } from './pdu_jstype_DisturbanceWind.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduDisturbanceWindConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DisturbanceWind } src
     */
    static to_cdr_body(writer, src) {
        PduVector3Converter.to_cdr_body(writer, src.value);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DisturbanceWind } dst
     * @returns { DisturbanceWind }
     */
    static cdr_body_to_js(reader, dst) {
        PduVector3Converter.cdr_body_to_js(reader, dst.value);
        return dst;
    }

    /**
     * @param { DisturbanceWind } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DisturbanceWind }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DisturbanceWind());
    }
}
